test/karma.config.mjs

```diff
@@ -0,0 +1,167 @@
+    throw new Error(`no browsers to launch, input: ${BROWSERS}`);
+  }
+  
+  config.set({
+    plugins: [
+      'karma-jasmine',
+      'karma-qunit',
+      'karma-chrome-launcher',
+      'karma-firefox-launcher',
+      'karma-coverage',
+      'karma-spec-reporter'
+    ],
+
+    // base path that will be used to resolve all patterns (eg. files, exclude)
+    basePath: '../',
+    autoWatch: false,
+    singleRun: true,
+    pingTimeout: CI ? 10000 : 2000,
+    browserDisconnectTimeout: CI ? 10000 : 2000,
+
+    // frameworks to use
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: [/*'jasmine',*/ 'qunit'],
+    browsers,
+    customLaunchers: {
+      ChromeCI: {
+        base: 'Chrome',
+        flags: [
+          '--no-sandbox',
+          '--disable-gpu',
+          '--disable-translate',
+          '--disable-extensions',
+          '--remote-debugging-port=9222',
+        ],
+      },
+      FirefoxCI: {
+        base: 'Firefox'
+      }
+    },
+
+    reporters: ['spec', 'coverage'],
+    coverageReporter: {
+      reporters: [
+        { type: 'lcov', dir: '.nyc_output/' },
+        // Karma uses subdirs by default to account for multiple browsers.
+        // For the JSON file, it's important we disable 'subdir' so that
+        // the 'nyc report' command can pick this up when combining code
+        // coverage with the Node.js test run.
+        { type: 'json', dir: '.nyc_output/', subdir: '.' }
+      ]
+    },
+
+    // list of files / patterns to load in the browser
+    files: [
+      { pattern: 'test/fixtures/**/*', included: false, served: true, watched: false, nocache: false },
+      { pattern: 'test/lib/*.js', included: true, served: true, watched: true, nocache: false },
+      { pattern: 'test/lib/tests.css', included: true, served: true, watched: true, nocache: false }, // qunit only
+      { pattern: 'test/visual/golden/**/*', included: false, served: true, watched: false, nocache: false },
+      { pattern: 'test/visual/assets/**/*', included: false, served: true, watched: false, nocache: false },
+
+      { pattern: 'dist/index.js', type: 'js', included: true, served: true, watched: true, nocache: true },
+      { pattern: 'dist/index.js.map', included: false, served: true, watched: false, nocache: true },
+
+      { pattern: 'test/karma.setup.js', type: 'js', included: true, served: true, watched: true, nocache: false },
+
+      // add test files last
+      ...(TEST_FILES ?
+        TEST_FILES.map(file => ({ pattern: file, type: 'js', included: true, served: true, watched: true, nocache: false })) :
+        [
+          process.env.TEST_SUITE === 'unit' && { pattern: 'test/unit/**/*.js', type: 'js', included: true, served: true, watched: true, nocache: false },
+          process.env.TEST_SUITE === 'visual' && { pattern: 'test/visual/**/*.js', type: 'js', included: true, served: true, watched: true, nocache: false },
+        ].filter((exists) => exists)
```

Comment:
I think there is no need to separate between suite any longer as well.
Browsers too since karma runs browsers in parallel
All in the follow up

Reply:
units vs visual is nice because it lets us separate concern on what are we testing.
Visual tests are more large scoped and probalby we need interaction tests too.
Different suites are good

Reply:
it doesn't mean you won't be able to run a single suite standalone.
It means that if you want to run 2 tests from different suites it won't spawn 2 servers/browsers etc.
And it might have other implications with code coverage